webauthn-rs-core/src/internals.rs

```diff
@@ -221,15 +221,31 @@ fn extensions_parser<T: Ceremony>(i: &[u8]) -> nom::IResult<&[u8], T::SignedExte
     )(i)
 }
 
-fn acd_parser(i: &[u8]) -> nom::IResult<&[u8], AttestedCredentialData> {
+fn aaguid_parser(i: &[u8]) -> nom::IResult<&[u8], Aaguid> {
+    // We have observed with Windows Hello on Windows 11 aaguids of 0 being
+    // serialized as 1 null-byte rather than 16 of them, as required by the spec
+    // (https://www.w3.org/TR/webauthn-2/#sctn-attested-credential-data).
+    if i[0] == 0 {
```

Comment:
We should probably make this `if Some(0) == i.get(0)` so we have the bounds check, else we may panic.